### PR TITLE
Update widgets roadmap

### PR DESCRIPTION
**Pull request by jdfreder**
Here's an update of the TODOs as they sit in my head.  Feel free to discuss..

cc @ellisonbg @sylvaincorlay

### CODE REVIEW

**Comment by jdfreder** [comment]
Whoops, looks like I managed to delete some things... Repairing that now.

**Comment by jdfreder** [comment]
Okay, fixed.

**Comment by jdfreder** [comment]
Anyone interested in discussing this?

File: README.md [hunk 1]
```diff
@@ -123,36 +123,53 @@ each milestone.
...
+* Hands off approach to styling.  ipywidgets only provides an API for
+  adding/removing CSS classes and the Style API.  The Style API provides the
+  ability to set layout CSS traits in a mutable fashion. All other styling APIs
   are deprecated.
-* Ability to run ipywidgets outside of the notebook.
-* Initial refactoring to start to use npm for packaging the JavaScript code.
-* A new jupyter-js-widgets repository, containing all of the JS of ipywidgets.  
-  ipywidgets itself will no longer contain any JS.
-* A new interact API that behaves like Sage's equivalent.
+* Include a new interact API that behaves like Sage's equivalent.
```
Comment (Carreau) [inline README.md:130]: I think there is already an issue for that ? link ? and also the `@interaction.options(...)` maybe.

File: README.md [hunk 2]
```diff
@@ -123,36 +123,53 @@ each milestone.
...
-  a modern transpiler, and offloading much of JavaScript weight lifting to third 
-  party packages.
-* Explore how we can start to treat widgets as regular output, and not have to have
-  a separate widget area in the DOM.
-* Explore how the existing codebase will integrate with phosphor. In particular, explore
-  using phosphor properties to abstract away the details of backbone.js models and replace
-  the backbone models with phosphorjs widget-based views.
+* Unit tests for jupyter-js-widgets individual widgets
+* Unit tests for widgetsnbextension, especially manager.js
+* Transition to ES6 or TypeScript.
```
Comment (Carreau) [inline README.md]: Gradual transition or total ? This looks like Widgets **will be** TS or ES6, while you can mix versions, and gradually improve. Also precise that user won't need to write in es6 or TS.

File: README.md [hunk 3]
```diff
@@ -164,7 +181,7 @@ each milestone.
 
 ## nbconvert
 
-### Next (4.2)
+### Next
```
Comment (Carreau) [inline README.md]: Next minor ? a blank next is ambiguous, as it evolves.

**Comment by Carreau** [comment]
You'll likely get some inut soon, we got some video meeting with GenePattern and Kbase folks at noon.

**Comment by ellisonbg** [comment]
I can have a look soon - took a week to dig out of last quarter and the dev meeting

File: README.md (hunk 3 above)
Comment (jdfreder) [inline README.md]: This was a rebase mistake, I shouldn't have edited this line.  Sorry!

File: README.md (hunk 2 above)
Comment (jdfreder) [inline README.md]: Sorry, gradual, will clarify.  Intent is **not** to rewrite the widgets

File: README.md (hunk 1 above)
Comment (jdfreder) [inline README.md:130]: Yeah, I think it was already discussed and the changes were made https://github.com/ipython/ipywidgets/pull/288 https://github.com/ipython/ipywidgets/issues/238

**Comment by SylvainCorlay** [comment]
I would like to add a few things for the near future roadmap.
- Regarding embeddable widgets, I would like to enable a mechanism for it to work with custom widget models and views to without having to manually put more with script tags on the page. We discussed multiple approaches during our meeting on the integration of a module version number so that we could compose npmcdn urls for custom models.
- Datetime and Date traits with serialization + Date picker and date time picker. (I already had prototypes). The date traits should work with the time-zone issues between browser and backend as discussed earlier.
- Use jupyter-js-outputarea for the ouput widget.

**Comment by jdfreder** [comment]
@SylvainCorlay sure, these all sound like good ideas.

**Comment by fperez** [comment]
ping @JamiesHQ

**Comment by fperez** [comment]
I manually merged the conflicts and added @SylvainCorlay's comments in a separate commit (0ac78734723ef01192a9da360b817331cf0026fb) to ensure that we capture the entire discussion. We can iterate further, but I'm trying to get the whole roadmap updated and digested...

**Pull request merged**